README.md

```diff
@@ -778,6 +786,36 @@ Show the detail statistic of specific entity type. You can specify which part of
 
 For example, `/lifetime creeper` shows all statistic of creeper in detail, and `/lifetime creeper removal` only shows removal statistic of creeper in detail 
 
+# Scarpet
+
+## Functions
+
+### `register_block(pos)`
+
+Registers a block position to be tracked with events by scarpet. It is added to a global tracker (which you can view with
+`registered()`). Blocks in this list will trigger events in `__on_block_event` if they occur, independently of the loggers.
+Returns `true` if block was not previously in list, and `false` if it was.
+
+### `deregister_block(pos)`
+
+Removes a block position from the list of block positions to be tracked. Returns `true` if block was previously in the
+list, and `false` if not.
+
+### `registered()`, `registered(pos)`
+
+Without any arguments, returns the list of block positions which are tracked by scarpet to trigger the block event events
+within scarpet. With a position, will return `true` or `false` based on whether or not that position is in the list.
+
+## Events
+
+### `__on_block_event(type, pos)`
```

Comment:
idk if there's a better name than `block_event` since it might confused with the piston block event thing

Reply:
ye, i wasn't sue what to add instead...

Reply:
cos in the code it's BaseEvent, but `__on_base_event` looks bad...

Reply:
u can give it another name like `microtiming_event`

Reply:
ye, great idea!